### DISCUSSION TITLE
Weird caching games when `--cc=tcc` is being used

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#226] [Q&A] [answered]
I have the following mapped keys in Vim for Nelua:

```vim
nmap <F5> :up <bar> !clear; /usr/bin/env -S nelua --cc=tcc % <CR>
nmap <S-F5> :up <bar> !clear; /usr/bin/env -S nelua --cc=tcc -SVt % <CR>
nmap <F6> :up <bar> !clear; /usr/bin/env nelua %<CR>
nmap <S-F6> :up <bar> !clear; /usr/bin/env -S nelua -SVt %<CR>
```

For some reason, when `nelua --cc=tcc -SVt` takes place, it produces the following output:

```
using config file '/home/stefanos/.config/nelua/neluacfg.lua'
startup      26.7 ms
parse        34.0 ms
preprocess   22.5 ms
analyze      217.6 ms
generate     28.5 ms
using cached generated /home/stefanos/.cache/nelua/inlinecasm.c
using cached binary /home/stefanos/.cache/nelua/inlinecasm
compile      0.6 ms
/home/stefanos/.cache/nelua/inlinecasm 
math.sqrt 16.002400 cycles
tcc compiler: true
run          1.4 ms
total time   331.2 ms

Press ENTER or type command to continue
```

but when the other command without the usage of `--cc=tcc` that defaults to GCC it produces the following output (wrapped here for reading purposes):

```
using config file '/home/stefanos/.config/nelua/neluacfg.lua'
startup      34.1 ms
parse        34.4 ms
preprocess   21.3 ms
analyze      216.6 ms
generate     28.3 ms
generated /home/stefanos/.cache/nelua/inlinecasm.c
gcc -x c "/home/stefanos/.cache/nelua/inlinecasm.c" 
    -x none -fwrapv -fno-strict-aliasing 
    -Wall -Wextra -fsanitize=address,undefined 
    -g -pipe -lm -o "/home/stefanos/.cache/nelua/inlinecasm"
compile      647.5 ms
/home/stefanos/.cache/nelua/inlinecasm 
math.sqrt 13.333800 cycles
tcc compiler: false
run          7.5 ms
total time   989.8 ms

Press ENTER or type command to continue
```

My question @edubart is: which behavior of the two outputs is the right one?

I forgot to mention that I first run the `F5` then the `shift-F5`, and afterwards the same process but with `F6` this time.

**Comment by edubart** [comment, marked as answer]
There is no difference when you use `-S` (sanitize option) with TCC, because it does not support sanitizer. So doing `nelua --cc=tcc -Svt` generates exactly the same code as `nelua --cc=tcc`, therefore it uses the cache.

However when you use GCC, the `-S` option will add `-fsanitize=address,undefined` to the compilation process, and have to recompile.

**Reply by stefanos82** [reply]
Ah good catch! I forgot TCC does not support sanitizing :confused:
